src/components/AppConfig/OpenAI.tsx

```diff
@@ -34,13 +37,15 @@ export function OpenAIConfig({ settings, secrets, secretsSet, onChange, onChange
   };
   return (
     <FieldSet label="OpenAI Settings">
-      <Field label="Use Azure OpenAI">
-        <Switch
-          name="useAzure"
-          data-testid={testIds.appConfig.useAzureOpenAI}
-          defaultChecked={settings.useAzure}
-          checked={settings.useAzure}
-          onChange={onChangeField}
+      <Field label="OpenAI Provider">
+        <Select
+          data-testid={testIds.appConfig.openAIProvider}
+          options={[
+            { label: 'OpenAI', value: 'openai' },
+            { label: 'Azure OpenAI', value: 'azure' },
+          ] as Array<SelectableValue<OpenAIProvider>>}
+          onChange={(e) => onChange({ ...settings, provider: e.value })}
```

Comment:
```suggestion
          ] as Array<SelectableValue<OpenAIProvider>>}
          value={settings.provider}
          onChange={(e) => onChange({ ...settings, provider: e.value })}
```

Missing a value field to display current selected value